app/main.py

```diff
@@ -2,36 +2,48 @@
 import pathlib
 import sys
 
-from basyx.aas import model
+from basyx.aas import model, adapter
 from basyx.aas.adapter import aasx
 
 from basyx.aas.backend.local_file import LocalFileObjectStore
 from basyx.aas.adapter.http import WSGIApp
 
-storage_path = os.getenv("STORAGE_PATH", "/storage")
-storage_type = os.getenv("STORAGE_TYPE", "AASX")
+storage_path = os.getenv("STORAGE_PATH", "storage")
+storage_type = os.getenv("STORAGE_TYPE", "LOCAL_FILE_READ_ONLY")
 base_path = os.getenv("API_BASE_PATH")
 
 wsgi_optparams = {}
 
 if base_path is not None:
     wsgi_optparams["base_path"] = base_path
 
-if storage_type == "LOCAL_FILE":
+if storage_type == "LOCAL_FILE_BACKEND":
     application = WSGIApp(LocalFileObjectStore(storage_path), aasx.DictSupplementaryFileContainer(), **wsgi_optparams)
 
-elif storage_type == "AASX":
+elif storage_type in "LOCAL_FILE_READ_ONLY":
     object_store: model.DictObjectStore = model.DictObjectStore()
     file_store: aasx.DictSupplementaryFileContainer = aasx.DictSupplementaryFileContainer()
 
-    for file in pathlib.Path(storage_path).glob("*.aasx"):
+    for file in pathlib.Path(storage_path).iterdir():
         if not file.is_file():
             continue
         print(f"Loading {file}")
-        with aasx.AASXReader(file) as reader:
-            reader.read_into(object_store=object_store, file_store=file_store)
+
+        if file.suffix.lower() == ".json":
+            with open(file) as f:
+                adapter.json.read_aas_json_file_into(object_store, f)
+        elif file.suffix.lower() == ".xml":
+            with open(file) as f:
+                adapter.xml.read_aas_xml_file_into(object_store, file)
+        elif file.suffix.lower() == ".aasx":
+            with aasx.AASXReader(file) as reader:
+                reader.read_into(object_store=object_store, file_store=file_store)
 
     application = WSGIApp(object_store, file_store, **wsgi_optparams)
 
 else:
-    print(f"STORAGE_TYPE must be either LOCAL_FILE or AASX! Current value: {storage_type}", file=sys.stderr)
+    print(f"STORAGE_TYPE must be either LOCAL_FILE or LOCAL_FILE_READ_ONLY! Current value: {storage_type}",
+          file=sys.stderr)
+
+# Lower the suffix
+# Lower the suffixes to also match the filenames with uppercase suffixes
```

Comment:
Seems like there's a missing new line at the end of the file. Other than that, it lgtm now.

Reply:
Resolved in https://github.com/rwth-iat/basyx-python-sdk-http-server-docker/pull/2/commits/b3b0750c05b5bb585da4308c9342de1100dc3bc3